src/main/java/join_room_use_case/JoinByIDInteractor.java

```diff
@@ -0,0 +1,39 @@
+package join_room_use_case;
+
+import signup_and_login.UserDBGateway;
+import signup_and_login.UserDBModel;
+
+import java.util.List;
+
+public class JoinByIDInteractor implements JoinByIDInputBoundary{
+    private final RoomDBGateway roomDBGateway;
+    private final UserDBGateway userDBGateway;
+    private final JoinByIDOutputBoundary presenter;
+
+    public JoinByIDInteractor(RoomDBGateway roomDBGateway, JoinByIDOutputBoundary presenter,
+                              UserDBGateway userDBGateway){
+        this.roomDBGateway = roomDBGateway;
+        this.userDBGateway = userDBGateway;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void handleJoinByID(JoinByIDRequestModel requestModel){
+        List<RoomDBModel> rooms = roomDBGateway.loadRooms();
+
+        for (RoomDBModel room : rooms) {
+            if (room.getName().equals(requestModel.getRoomName())){
+                List<UserDBModel> users = userDBGateway.loadUsers();
+                for (UserDBModel user : users){
+                    if (user.getUserID() == requestModel.getCurrUserID()){
+                        roomDBGateway.updateRoomActiveUsers(user);
+                        presenter.prepareRoomView(room.getRoomID());
+                    }
+                } presenter.prepareFailView("No Such User Found!");
+            }
+        }
+        presenter.prepareFailView("No Such Room Found!");
+        }
+
```

Comment:
has duplicated logic for preparing failure views when no user or room is found. . Consider refactoring this part into a separate method to reduce duplication.

Reply:
see 261bbb3d9b786d14579a492442aee83ca2b80cd1